Sources_v3/StreamChat/Controllers/CurrentUserController/CurrentUserController.swift

```diff
@@ -364,6 +364,47 @@ public extension _CurrentChatUserController {
         }
     }
     
+    /// Updates the current user.
+    ///
+    /// By default all data is `nil`, and it won't be updated unless a value is provided.
+    ///
+    /// - Parameters:
+    ///   - name: Optionally provide a new name to be updated.
+    ///   - imageURL: Optionally provide a new image to be updated.
+    ///   - userExtraData: Optionally provide new user extra data to be updated.
+    ///   - completion: Called when user is successfuly updated, or with error.
+    func updateUser(
+        name: String? = nil,
+        imageURL: URL? = nil,
+        userExtraData: ExtraData.User? = nil,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        guard let currentUserId = currentUser?.id else {
+            completion?(ClientError.CurrentUserDoesNotExist())
+            return
+        }
+        
```

Comment:
We should probably also add a check that at least one of the fields is not `nil`. In that case we can just call `completion(nil)` but we should log a warning that nothing happened.